modules/localization/msf/local_pyramid_map/pyramid_map/pyramid_map_matrix.cc

```diff
@@ -0,0 +1,1640 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+#include "modules/localization/msf/local_map/pyramid_map/pyramid_map_matrix.h"
+
+#include <algorithm>
+
+namespace apollo {
+namespace localization {
+namespace msf {
+
+PyramidMapMatrix::PyramidMapMatrix() { Clear(); }
+
+PyramidMapMatrix::~PyramidMapMatrix() { Clear(); }
+
+PyramidMapMatrix::PyramidMapMatrix(const PyramidMapMatrix& map_matrix) {
+  Clear();
+  resolution_num_ = map_matrix.resolution_num_;
+  ratio_ = map_matrix.ratio_;
+  rows_mr_ = map_matrix.rows_mr_;
+  cols_mr_ = map_matrix.cols_mr_;
+  ratio_multiplier_ = map_matrix.ratio_multiplier_;
+
+  intensity_matrixes_ = map_matrix.intensity_matrixes_;
+  intensity_var_matrixes_ = map_matrix.intensity_var_matrixes_;
+  altitude_matrixes_ = map_matrix.altitude_matrixes_;
+  altitude_var_matrixes_ = map_matrix.altitude_var_matrixes_;
+  ground_altitude_matrixes_ = map_matrix.ground_altitude_matrixes_;
+  count_matrixes_ = map_matrix.count_matrixes_;
+  ground_count_matrixes_ = map_matrix.ground_count_matrixes_;
+
+  has_intensity_ = map_matrix.has_intensity_;
+  has_intensity_var_ = map_matrix.has_intensity_var_;
+  has_altitude_ = map_matrix.has_altitude_;
+  has_altitude_var_ = map_matrix.has_altitude_var_;
+  has_ground_altitude_ = map_matrix.has_ground_altitude_;
+  has_count_ = map_matrix.has_count_;
+  has_ground_count_ = map_matrix.has_ground_count_;
+}
+
+void PyramidMapMatrix::Init(const BaseMapConfig& config) {
+  const PyramidMapConfig* pconfig =
+      dynamic_cast<const PyramidMapConfig*>(&config);
+  Init(pconfig->map_node_size_y_, pconfig->map_node_size_x_,
+       pconfig->has_intensity_, pconfig->has_intensity_var_,
+       pconfig->has_altitude_, pconfig->has_altitude_var_,
+       pconfig->has_ground_altitude_, pconfig->has_count_,
+       pconfig->has_ground_count_, pconfig->resolution_num_,
+       pconfig->resolution_ratio_);
+}
+
+void PyramidMapMatrix::Reset() {
+  for (unsigned int i = 0; i < resolution_num_; i++) {
+    Reset(i);
```

Comment:
what does this try to reset?

Reply:
It trys to reset each resolution level matrix